src/parser/pegen.ts

```diff
@@ -1983,8 +1989,12 @@ export function get_names(p: Parser, names_with_defaults: NameDefaultPair[] | nu
 //     }
 //     return seq;
 // }
+type exprOrNull<T> = T extends NameDefaultPair<infer R> ? (R extends expr ? R : expr | null) : expr | null;
 
-export function get_defaults(p: Parser, names_with_defaults: NameDefaultPair[]): exprOrNone[] {
+export function get_defaults<T extends NameDefaultPair<exprOrNull<T>>>(
+    p: Parser,
+    names_with_defaults: T[]
+): exprOrNull<T>[] {
     return names_with_defaults.map((pair) => pair.value);
 }
 
```

Comment:
This also was 🤯 for me... It means that typescript knows if you're passing a nameDefaultPare with `expr` only values or `expr | null` values. (kw_defaults have `expr | null`, defaults have only `expr`).